orca-front50/src/main/groovy/com/netflix/spinnaker/orca/front50/pipeline/PipelineStage.java

```diff
@@ -65,8 +68,18 @@ public void taskGraph(Stage stage, TaskNode.Builder builder) {
   @Override
   public void prepareStageForRestart(Stage stage) {
     stage.getContext().remove("status");
-    stage.getContext().remove("executionName");
-    stage.getContext().remove("executionId");
+    if (!stage
```

Comment:
nit: i, personally, find this really hard to read. I would maybe rewrite as:

```java
StageContext context = (StageContext)stage.getContext();
boolean restartChild = (boolean)(context.getCurrentOnly("_skipPipelineRestart", false));

if (restartChild) {
  ..
 } else {
  ..
}
```

Note the use of `getCurrentOnly` - that way you are only inspecting the context of the current stage not any of the outputs of its parents (a bug that we've run into a bunch)